ec/ecresource/deploymentresource/elasticsearch/v2/elasticsearch.go

```diff
@@ -0,0 +1,413 @@
+
+	res.Plan.Elasticsearch, ds = ElasticsearchConfigPayload(ctx, es.Config, res.Plan.Elasticsearch)
+	diags.Append(ds...)
+
+	diags.Append(elasticsearchSnapshotSourcePayload(ctx, es.SnapshotSource, res.Plan)...)
+
+	diags.Append(elasticsearchExtensionPayload(ctx, es.Extension, res.Plan.Elasticsearch)...)
+
+	if es.Autoscale.Value != "" {
+		autoscaleBool, err := strconv.ParseBool(es.Autoscale.Value)
+		if err != nil {
+			diags.AddError("failed parsing autoscale value", err.Error())
+		} else {
+			res.Plan.AutoscalingEnabled = &autoscaleBool
+		}
+	}
+
+	res.Settings, ds = ElasticsearchTrustAccountPayload(ctx, es.TrustAccount, res.Settings)
+	diags.Append(ds...)
+
+	res.Settings, ds = ElasticsearchTrustExternalPayload(ctx, es.TrustExternal, res.Settings)
+	diags.Append(ds...)
+
+	elasticsearchStrategyPayload(es.Strategy, res.Plan)
+
+	return res, diags
+}
+
+func (es *ElasticsearchTF) topologiesPayload(ctx context.Context, topologies []*models.ElasticsearchClusterTopologyElement) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	diags.Append(topologyPayload(ctx, es.HotContentTier, "hot_content", topologies)...)
+	diags.Append(topologyPayload(ctx, es.CoordinatingTier, "coordinating", topologies)...)
+	diags.Append(topologyPayload(ctx, es.MasterTier, "master", topologies)...)
+	diags.Append(topologyPayload(ctx, es.WarmTier, "warm", topologies)...)
+	diags.Append(topologyPayload(ctx, es.ColdTier, "cold", topologies)...)
+	diags.Append(topologyPayload(ctx, es.FrozenTier, "frozen", topologies)...)
+	diags.Append(topologyPayload(ctx, es.MlTier, "ml", topologies)...)
+
+	return diags
+}
+
+func topologyPayload(ctx context.Context, topologyObj types.Object, id string, topologies []*models.ElasticsearchClusterTopologyElement) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	if !topologyObj.IsNull() && !topologyObj.IsUnknown() {
+		var topology ElasticsearchTopologyTF
+
+		ds := tfsdk.ValueAs(ctx, topologyObj, &topology)
+		diags.Append(ds...)
+
+		if !ds.HasError() {
+			diags.Append(topology.Payload(ctx, id, topologies)...)
+		}
+	}
+
+	return diags
+}
+
+func (es *Elasticsearch) setTopology(topologies ElasticsearchTopologies) {
+	set := topologies.Set()
+
+	for id, topology := range set {
+		topology := topology
+		switch id {
+		case "hot_content":
+			es.HotTier = &topology
+		case "coordinating":
+			es.CoordinatingTier = &topology
+		case "master":
+			es.MasterTier = &topology
+		case "warm":
+			es.WarmTier = &topology
+		case "cold":
+			es.ColdTier = &topology
+		case "frozen":
+			es.FrozenTier = &topology
+		case "ml":
+			es.MlTier = &topology
+		}
+	}
+}
+
+func unsetElasticsearchCuration(payload *models.ElasticsearchPayload) {
+	if payload.Plan.Elasticsearch != nil {
+		payload.Plan.Elasticsearch.Curation = nil
+	}
+
+	if payload.Settings != nil {
+		payload.Settings.Curation = nil
+	}
+}
+
+func UpdateNodeRolesOnDedicatedTiers(topologies []*models.ElasticsearchClusterTopologyElement) {
```

Comment:
Doesn't need to be exported 

```suggestion
func updateNodeRolesOnDedicatedTiers(topologies []*models.ElasticsearchClusterTopologyElement) {
```